enclave/core/sgx/seal.c

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/string.h>
+#include <openenclave/enclave.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/safecrt.h>
+#include "report.h"
+
+#define _OE_MAX_SEAL_PLUGIN 16
+
+static const oe_seal_plugin_definition_sgx_t* _plugins[_OE_MAX_SEAL_PLUGIN];
+static int _num_plugins = 0;
+
+oe_result_t oe_register_seal_plugin(
+    const oe_seal_plugin_definition_sgx_t* plugin,
```

Comment:
As discussed in SIG-Arch meeting, change to void* and move to non-sgx directory.
It's fine to also have an sgx-specific inline wrapper if you want, in the sgx .h file, called oe_register_sgx_seal_plugin or oe_register_seal_plugin_sgx.